src/dune_file_watcher/dune_file_watcher.ml

```diff
@@ -582,6 +588,68 @@ let create_fsevents ?(latency = 0.2) ~(scheduler : Scheduler.t) () =
   ; sync_table
   }
 
+let fswatch_win_exclude_patterns =
+  List.map ~f:Re.compile
+    [ Re.str "/_opam"
+    ; Re.str "/_esy"
+    ; Re.str
+        "/.#" (* Such files can be created by Emacs and also Dune itself. *)
+    ; Re.seq [ Re.char '~'; Re.eos ]
+    ; Re.str "/#"
+    ; Re.str "/4913" (* https://github.com/neovim/neovim/issues/3460 *)
+    ; Re.str "/.git"
+    ; Re.str "/.hg"
+    ; Re.seq [ Re.bos; Re.any; Re.str ":/windows" ]
+    ]
+
+let fswatch_win_filter path =
+  let path =
+    String.concat ~sep:"/"
+      (String.split_on_char ~sep:'\\'
+         (String.lowercase_ascii (Path.to_string path)))
+  in
+  List.for_all
+    ~f:(fun re -> not (Re.execp re path))
+    fswatch_win_exclude_patterns
```

Comment:
How come you didn't compile it into a single re with `Re.alt`?

Reply:
Good point, fixed.